### PR TITLE
fix: remove occurences of the futurely deprecated title struct

### LINKED ISSUE #740: Edit code to not use Title struct that will be deprecated in future
There are apparently several occurences of the `Title` struct in the website. We should not recommend people to use a struct that will be deprecated. I could do this

### PR DESCRIPTION
**Pull request by janTatesa**
This issue fixes #740

### CODE REVIEW

**Review by orhun — APPROVED** [review]

File: code/templates/async-template-counter/src/components/fps.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ impl Component for FpsCounter {
     let rect = rects[0];
 
     let s = format!("{:.2} fps (app) {:.2} fps (render)", self.app_fps, self.render_fps);
-    let block = Block::default().title(block::Title::from(s.dim()).alignment(Alignment::Right));
+    let block = Block::default().title(Line::from(s.dim()).alignment(Alignment::Right));
```
Comment (joshka) [inline code/templates/async-template-counter/src/components/fps.rs]:
```suggestion
    let block = Block::default().title(Line::from(s).dim().right_aligned());
```

File: code/tutorials/counter-app-basic/src/main.rs [hunk 2]
```diff
@@ -85,22 +82,18 @@ impl App {
...
-        ]));
+        ]);
         let block = Block::bordered()
             .title(title.alignment(Alignment::Center))
-            .title(
-                instructions
-                    .alignment(Alignment::Center)
-                    .position(Position::Bottom),
-            )
+            .title_bottom(instructions.alignment(Alignment::Center))
```
Comment (joshka) [inline code/tutorials/counter-app-basic/src/main.rs]:
```suggestion
            .title_bottom(instructions.centered())
```

File: code/tutorials/counter-app-error-handling/src/main.rs [hunk 3]
```diff
@@ -114,22 +111,18 @@ impl App {
...
-        ]));
+        ]);
         let block = Block::default()
             .title(title.alignment(Alignment::Center))
-            .title(
-                instructions
-                    .alignment(Alignment::Center)
-                    .position(Position::Bottom),
-            )
+            .title_bottom(instructions.alignment(Alignment::Center))
```
Comment (joshka) [inline code/tutorials/counter-app-error-handling/src/main.rs]:
```suggestion
            .title_bottom(instructions.centered())
```

File: src/content/docs/recipes/widgets/block.md [hunk 4]
```diff
@@ -30,13 +30,13 @@ let b = Block::default()
...
+You can also use the `Line` struct for better positioning or multiple titles.
 
 ```rust
 let b = Block::default()
-    .title(block::Title::from("Left Title").alignment(Alignment::Left))
-    .title(block::Title::from("Middle Title").alignment(Alignment::Center))
-    .title(block::Title::from("Right Title").alignment(Alignment::Right))
+    .title(Line::from("Left Title").alignment(Alignment::Left))
+    .title(Line::from("Middle Title").alignment(Alignment::Center))
+    .title(Line::from("Right Title").alignment(Alignment::Right))
```
Comment (joshka) [inline src/content/docs/recipes/widgets/block.md]:
```suggestion
    .title(Line::from("Left Title").left_aligned())
    .title(Line::from("Middle Title").centered())
    .title(Line::from("Right Title").right_aligned())
```

**Review by joshka — CHANGES_REQUESTED** [review]
The examples should generally be copied verbatim from the ratatui repo, so that they stay in sync with what we're releasing. The process for this is a bit more complex than a standard PR approach and is documented in the examples folder [README](https://github.com/ratatui/ratatui-website/blob/main/code/examples/ratatui-examples/README.md). So for now, remove these files from this PR and I'll do the fixup based on the syncing the examples instead.

Other than that, I've added some nit suggestions to simplify the code a touch more.

**Comment by janTatesa** [comment]
Are you aware of my changes?

**Comment by joshka** [comment]
> Are you aware of my changes?

The first comment in the review is the part that matters here the most "The examples should generally be copied verbatim from the ratatui repo,"

I may have been a bit unclear about what exactly I meant by this though, so my apologies. We don't edit the example code in this repo. It is authored in the ratatui repo and then copied into this repo sometime after each release so that the website code aligns with the latest release (although sometimes there can be some post-release updates). The process for doing that is a bit complex as I prefer to graft the history from the other repo into this one rather than having them disconnected. This is documented in the [examples README](https://github.com/ratatui/ratatui-website/blob/main/code/examples/ratatui-examples/README.md). Because of this complexity it's usually something I do as it can be fragile to moves / deletions / refactoring etc. (so the instructions are likely not 100% correct).

**Comment by janTatesa** [comment]
I've meant that I deleted the examples and accepted your changes

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
> I've meant that I deleted the examples and accepted your changes

Ah got it - I just re-read the PR changes. Apologies. I missed that the examples were no longer a part of this.

**Pull request merged**